### PR TITLE
Polish project accessors

### PR DESCRIPTION
**Pull request by melix**
Supercedes https://github.com/gradle/gradle/pull/16538 after rebase

### CODE REVIEW

**Comment by melix** [comment]
@bot-gradle test RFM

File: subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/typesafe/AbstractTypeSafeProjectAccessorsIntegrationTest.groovy [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import org.gradle.integtests.fixtures.AbstractHttpDependencyResolutionTest
+import org.gradle.integtests.fixtures.BuildOperationsFixture
+import org.gradle.integtests.fixtures.FeaturePreviewsFixture
+
+abstract class AbstractTypeSafeProjectAccessorsIntegrationTest extends AbstractHttpDependencyResolutionTest {
+
+    final BuildOperationsFixture operations = new BuildOperationsFixture(executer, testDirectoryProvider)
+
+    def setup() {
+        FeaturePreviewsFixture.enableTypeSafeProjectAccessors(settingsFile)
```
Comment (ljacomet) [inline subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/typesafe/AbstractTypeSafeProjectAccessorsIntegrationTest.groovy:28]: Dang forgot to use that fixture in other tests I fixed ...

File: subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/declaring_dependencies.adoc [hunk 2]
```diff
@@ -293,6 +293,39 @@ include::sample[dir="snippets/dependencyManagement/declaringDependencies-project
...
 
+[[sec:type-safe-project-accessors]]
+==== Type-safe project dependencies
+
+[WARNING]
+====
+Type-safe dependency accessors are an incubating feature which must be enabled explicitly.
+Implementation may change at any time.
+To add support for type-safe project accessors, add this to your `settings.gradle(.kts)` file:
+```
```
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/declaring_dependencies.adoc:304]: You are missing the closing part of that block

**Review by ljacomet — CHANGES_REQUESTED** [review]
Looks good overall!
Nice to have broader support for project names.

Fix request in the docs:

**Comment by melix** [comment]
@bot-gradle test and merge

**Comment by melix** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
I'm taking care of this. Please don't trigger new builds.

**Comment by melix** [comment]
> I'm taking care of this. Please don't trigger new builds.

Thanks @blindpirate !

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Pull request merged**